### ISSUE TITLE
Allow user impersonation to quickly try out other roles

### ISSUE THREAD

**Issue by sschuberth** [eclipse-apoapsis/ort-server#1012]
Before inviting new user or changing roles of a user, you want to be very sure to not accidentally grant to wide permissions. To double-check the planned roles to give, an "impersonation" feature would be great that allows to temporarily play "what if" by giving one's own user different roles to see the effect of the role change.

**Comment by mnonnenmacher** [comment]
Keycloak has an impersonation feature: https://www.keycloak.org/docs/latest/server_admin/#con-user-impersonation_server_administration_guide
I haven't tried it, but maybe this can be used without any code changes in the server.

**Comment by sschuberth** [comment]
> Keycloak has an impersonation feature

It seems to be [available in version 24.0.0](https://www.keycloak.org/docs/24.0.0/server_admin/#con-user-impersonation_server_administration_guide) already, so that's good.

**Comment by mnonnenmacher** [comment]
@sschuberth Did you check if this works for you or if this issue is still required?

**Comment by sschuberth** [comment]
No, I didn't check. In any case, I'd expect support for this to be implemented in the ORT Server UI directly, probably from the menu for the profile icon.

**Comment by mnonnenmacher** [comment]
There is an endpoint for impersonation:
https://www.keycloak.org/docs-api/latest/rest-api/index.html#_post_adminrealmsrealmusersuser_idimpersonation

But I could not find documentation for how to end the impersonation and switch back to the original user.
